src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1101,15 +1101,21 @@ private static bool IsWindowsApplication(string fileName)
 #if UNIX
             return false;
 #else
-            if (!Platform.IsWindowsDesktop) { return false; }
+            if (!Platform.IsWindowsDesktop)
+            {
+                return false;
+            }
 
-            // SHGetFileInfo() does not understand reparse points and returns 0 ("non exe or error")
-            // so we are trying to get a real path before.
-            // It is a workaround for Microsoft Store applications.
-            string realPath = Microsoft.PowerShell.Commands.InternalSymbolicLinkLinkCodeMethods.WinInternalGetTarget(fileName);
-            if (realPath is not null)
+            // The function 'SHGetFileInfo()' does not understand reparse points and returns 0 ("non exe or error")
+            // for a symbolic link file, so we try to get the immediate link target in that case.
+            // Why not get the final target (use 'returnFinalTarget: true')? Because:
+            //  1. When starting a process on Windows, if the 'FileName' is a symbolic link, the immediate link target will automatically be used,
+            //     but the OS does not do recursive resolution when the immediate link target is also a symbolic link.
+            //  2. Keep the same behavior as before adopting the 'LinkTarget' and 'ResolveLinkTarget' APIs in .NET 6.
+            string linkTarget = new FileInfo(fileName).ResolveLinkTarget(returnFinalTarget: false)?.FullName;
```

Comment:
```suggestion
            string linkTarget = Directory.ResolveLinkTarget(fileName, returnFinalTarget: false)?.FullName;
```

Reply:
Cannot use that API. It pass in `isDirectory: true` when calling `FileSystem.ResolveLinkTarget`, while `fileName` is not a directory.

Reply:
There is File.ResolveLinkTarget()

Reply:
I was looking for that one! Thanks :)